src/ctapipe/calib/camera/extractor.py

```diff
@@ -0,0 +1,164 @@
+"""
+Extraction algorithms to compute the statistics from a chunk of images
+"""
+
+__all__ = [
+    "StatisticsExtractor",
+    "PlainExtractor",
+    "SigmaClippingExtractor",
+]
+
+from abc import abstractmethod
+
+import numpy as np
+from astropy.stats import sigma_clipped_stats
+
+from ctapipe.containers import StatisticsContainer
+from ctapipe.core import TelescopeComponent
+from ctapipe.core.traits import Int
+
+
+class StatisticsExtractor(TelescopeComponent):
+    """
+    Base component to handle the extraction of the statistics from a table
+    containing e.g. charges, peak times and/or charge variances (images).
+    """
+
+    chunk_size = Int(
+        2500,
+        help="Size of the chunk used for the calculation of the statistical values",
+    ).tag(config=True)
+
+    def __call__(
+        self,
+        table,
+        masked_pixels_of_sample=None,
+        chunk_shift=None,
+        col_name="image",
+    ) -> dict:
```

Comment:
Wouldn't a `Table` be a more suitable return value?

Reply:
I think instead of timestamp, it should be `timestamp_start`, `timestamp_stop`, and maybe also add the event_ids of the first / last events of the chunk?

Reply:
Before we had `extraction_start` and `extraction_stop` which is more descriptive I think. Only the `event_ids` enough or do we need also the`obs_ids` (and `tel_ids`)?

Reply:
Resolved with the latest commit. I choose `time_start`, `time_end` and `event_id_start`,  `event_id_end`.